hips/draw/healpix.py

```diff
@@ -88,3 +95,115 @@ def healpix_to_hips(hpx_data, tile_width, base_path, file_format='fits'):
 
     properties = HipsSurveyProperties(data=data)
     properties.write(base_path / 'properties')
+
+def hips_to_healpix_array(order: int):
+    """Create array giving mapping between:
+
+    HEALPix index ==> XY index
+    XY index ==> HEALPix index
+
+    Parameters
+    ----------
+    order : int
+        Order of HEALPix array.
+    """
+    def fill_array(npix, nsize, pos):
+        size = nsize ** 2
+        npix_idx = [[0 for i in range(size // 4)] for j in range(4)]
+        nb = [0 for i in range(4)]
+
+        for i in range(size):
+            if (i % nsize) < (nsize // 2):
+                dg = 0
+            else:
+                dg = 1
+
+            if i < (size / 2):
+                bh = 1
+            else:
+                bh = 0
+
+            quad = (dg << 1) | bh
+
+            if pos is None:
+                j = i
+            else:
+                j = pos[i]
+
+            npix[j] = npix[j] << 2 | quad
+            npix_idx[quad][nb[quad]] = j
+            nb[quad] += 1
+
+        if size > 4:
+            for i in range(4):
+                fill_array(npix, nsize // 2, npix_idx[i])
+
+    if order == 0:
+        xy_to_healpix = healpix_to_xy = [0]
+        return
+
+    nsize = 2 ** order
+    xy_to_healpix = [0 for i in range(nsize ** 2)]
+    healpix_to_xy = [0 for i in range(nsize ** 2)]
+
+    fill_array(xy_to_healpix, nsize, None)
+
+    for i in range(len(xy_to_healpix)):
+        healpix_to_xy[xy_to_healpix[i]] = i
+
+    return healpix_to_xy, xy_to_healpix
+
+def hips_to_healpix(hips_url: str, npix: int, hpx_output_path: str):
```

Comment:
Let's not mix computation with I/O.

I would suggest to just take the HEALPix pixels as a Numpy array:
https://hips.readthedocs.io/en/latest/api/hips.healpix_to_hips.html

This assumes that the HEALPix fits in memory, but that's the common use case, and supporting larger images requires fancy file I/O where only chunks of the image are memmapped. Let's not do that.